lib/Models/Terria.ts

```diff
@@ -2102,22 +2104,32 @@ async function interpretStartData(
           version: isJsonString(startData.version)
             ? startData.version
             : SHARE_VERSION,
-          initSources: isJsonObjectArray(startData.initSources)
-            ? startData.initSources
+          initSources: Array.isArray(startData.initSources)
+            ? (startData.initSources.filter(
+                initSource =>
+                  isJsonString(initSource) || isJsonObject(initSource)
+              ) as (string | JsonObject)[])
             : []
         };
       }
 
       if (startDataV8 !== null && Array.isArray(startDataV8.initSources)) {
+        // Push startData to initSources array
         runInAction(() => {
           terria.initSources.push(
-            ...startDataV8!.initSources.map((initSource: unknown) => {
-              return {
-                name,
-                data: isJsonObject(initSource, false) ? initSource : {},
-                errorSeverity
-              };
-            })
+            ...startDataV8!.initSources.map(initSource =>
+              isJsonString(initSource)
+                ? {
+                    name,
+                    initUrl: initSource,
```

Comment:
Whats the difference here, why are we storing an `initUrl` when its a json string, and a `data` when its not?

Reply:
I have added some comments to make it more clear

https://github.com/TerriaJS/terriajs/blob/8a94ac1002bfe29024092be3ccd4c0885c29581a/lib/Models/Terria.ts#L2157-L2178

The `InitSource` stuff is a bit weird - and most of the property names are coming from our JSON config - so we can't really change them